src/core/MOM_porous_barriers.F90

```diff
@@ -0,0 +1,166 @@
+       eta_s, & ! layer height used for fit [Z ~> m]
+       eta_prev ! interface height of previous layer [Z ~> m]
+  isd = G%isd; ied = G%ied; jsd = G%jsd; jed = G%jed
+  IsdB = G%IsdB; IedB = G%IedB; JsdB = G%JsdB; JedB = G%JedB
+
+  !eta is zero at surface and decreases downward
+  !all calculations are done in [m]
+
+  nk = SZK_(G)
+
+  !currently no treatment for using optional find_eta arguments if present
+  call find_eta(h, tv, G, GV, US, eta)
+
+  do I=IsdB,IedB; do j=jsd,jed
+     if (G%porous_DavgU(I,j) < 0.) then
+        do K = nk+1,1,-1
+           eta_s = max(US%Z_to_m*eta(I,j,K), US%Z_to_m*eta(I+1,j,K)) !take shallower layer height
+           !eta_s = 0.5 * (US%Z_to_m*eta(I,j,K) + US%Z_to_m*eta(I+1,j,K)) !take arithmetic mean
+           if (eta_s <= G%porous_DminU(I,j)) then
+              pbv%por_layer_widthU(I,j,K) = 0.0
+              A_layer_prev = 0.0
+              if (K < nk+1) then
+                 pbv%por_face_areaU(I,j,k) = 0.0; endif
+           else
+              call calc_por_layer(G%porous_DminU(I,j), G%porous_DmaxU(I,j), G%porous_DavgU(I,j),&
+                   eta_s, w_layer, A_layer)
+              pbv%por_layer_widthU(I,j,K) = w_layer
+              if (k <= nk) then
+                 if ((eta_s - eta_prev) > 0.0) then
+                   pbv%por_face_areaU(I,j,k) = (A_layer - A_layer_prev)/&
+                   (eta_s-eta_prev)
+                 else
+                   pbv%por_face_areaU(I,j,k) = 0.0; endif
+              endif
+               eta_prev = eta_s
+               A_layer_prev = A_layer
+           endif; enddo
+       endif; enddo; enddo
+
+  do i=isd,ied; do J=JsdB,JedB
+     if (G%porous_DavgV(i,J) < 0.) then
+        do K = nk+1,1,-1
+           eta_s = max(US%Z_to_m*eta(i,J,K), US%Z_to_m*eta(i,J+1,K)) !take shallower layer height
+           !eta_s = 0.5 * (US%Z_to_m*eta(i,J,K) + US%Z_to_m*eta(i,J+1,K)) !take arithmetic mean
+           if (eta_s <= G%porous_DminV(i,J)) then
+              pbv%por_layer_widthV(i,J,K) = 0.0
+              A_layer_prev = 0.0
+              if (K < nk+1) then
+                 pbv%por_face_areaV(i,J,k) = 0.0; endif
+           else
+              call calc_por_layer(G%porous_DminV(i,J), G%porous_DmaxV(i,J), G%porous_DavgV(i,J),&
+                   eta_s, w_layer, A_layer)
+              pbv%por_layer_widthV(i,J,K) = w_layer
+              if (k <= nk) then
+                 if ((eta_s - eta_prev) > 0.0) then
+                   pbv%por_face_areaV(i,J,k) = (A_layer - A_layer_prev)/&
+                   (eta_s-eta_prev)
+                 else
+                   pbv%por_face_areaU(I,j,k) = 0.0; endif
+              endif
+               eta_prev = eta_s
+               A_layer_prev = A_layer
+           endif; enddo
+       endif; enddo; enddo
+
+end subroutine por_widths
+
+subroutine calc_por_layer(D_min, D_max, D_avg, eta_layer, w_layer, A_layer)
+!subroutine to calculate the profile fit for a layer
```

Comment:
This comment needs to use the `!<` syntax, or better yet, put the description above the `subroutine` and use `!>`.